bluemira/builders/coil_supports.py

```diff
@@ -0,0 +1,252 @@
+
+from bluemira.base.builder import Builder
+from bluemira.base.components import Component, PhysicalComponent
+from bluemira.base.error import BuilderError
+from bluemira.base.parameter_frame import Parameter, ParameterFrame
+from bluemira.display.palettes import BLUE_PALETTE
+from bluemira.geometry.coordinates import Coordinates
+from bluemira.geometry.face import BluemiraFace
+from bluemira.geometry.plane import BluemiraPlane
+from bluemira.geometry.tools import (
+    boolean_cut,
+    boolean_fuse,
+    extrude_shape,
+    make_polygon,
+    slice_shape,
+)
+from bluemira.geometry.wire import BluemiraWire
+
+
+@dataclass
+class ITERGravitySupportBuilderParams(ParameterFrame):
+    """
+    ITER-like gravity support parameters
+    """
+
+    x_g_support: Parameter[float]
+    z_gs: Parameter[float]
+    tf_wp_width: Parameter[float]
+    tf_wp_depth: Parameter[float]
+    tk_tf_side: Parameter[float]
+    tf_gs_tk_plate: Parameter[float]
+    tf_gs_g_plate: Parameter[float]
+    tf_gs_base_depth: Parameter[float]
+
+
+class ITERGravitySupportBuilder(Builder):
+    """
+    ITER-like gravity support builder
+
+    Parameters
+    ----------
+    params:
+        Parameters to use
+    build_config:
+        Build config to use
+    tf_kz_keep_out_zone:
+        TF coil wire keep-out-zone for the outer edge of the TF coil (including casing)
+        Note that this should be on the y=0 plane.
+    """
+
+    param_cls: Type[ITERGravitySupportBuilderParams] = ITERGravitySupportBuilderParams
+
+    def __init__(
+        self,
+        params: Union[ITERGravitySupportBuilderParams, Dict],
+        build_config: Dict,
+        tf_xz_keep_out_zone: BluemiraWire,
+    ):
+        super().__init__(params, build_config)
+        self.tf_xz_keep_out_zone = tf_xz_keep_out_zone
+
+    def build(self) -> Component:
+        """
+        Build the ITER-like gravity support component.
+        """
+        xyz = self.build_xyz()
+        return self.component_tree([self.build_xz(xyz)], self.build_xy(), [xyz])
+
+    def build_xz(self, xyz_component):
+        """
+        Build the x-z component of the ITER-like gravity support.
+        """
+        xz_plane = BluemiraPlane((0, 0, 0), (0, 1, 0))
+        slice_result = slice_shape(xyz_component.shape, xz_plane)
+
+        # Process UGLY SLICE
+        wires = sorted(slice_result, key=lambda wire: wire.length)
+        wire_list = [wires.pop()]
+        wire_list.extend(wires)
+        shape = BluemiraFace(wire_list)
+
+        component = PhysicalComponent("ITER-like gravity support", shape)
+        component.display_cad_options.color = BLUE_PALETTE["TF"][2]
+        component.plot_options.face_options["color"] = BLUE_PALETTE["TF"][2]
+        return component
+
+    def build_xy(self):
+        """
+        Build the x-y component of the ITER-like gravity support.
+        """
+        pass
+
+    def _get_intersection_wire(self, width):
+        x_inner_line = self.params.x_g_support - 0.5 * width
```

Comment:
We moved away from using parameters like this to using `.value` this will need to be changed once the parameterframe is setup properly